### PR TITLE
CRM-12252 Add in help_pre and help_post fields for price field options

### PR DESCRIPTION
**Pull request by seamuslee001**
- [CRM-12252: What is price set price option description for?](https://issues.civicrm.org/jira/browse/CRM-12252)

### CODE REVIEW

File: CRM/Price/BAO/PriceField.php [hunk 1]
```diff
@@ -376,21 +376,25 @@ public static function addQuickFormElement(
           $taxAmount = CRM_Utils_Array::value('tax_amount', $opt);
           if ($field->is_display_amounts) {
             $opt['label'] = !empty($opt['label']) ? $opt['label'] . '&nbsp;-&nbsp;' : '';
+            $pre_help = isset($opt['help_pre']) ?
+              '<span class="crm-price-amount-help-pre">' . $opt['help_pre'] . '</span>: ' : '';
+            $post_help = isset($opt['help_post']) ?
+              ': <span class="crm-price-amount-help-post">' . $opt['help_post'] . '</span>' : '';
```
Comment (jitendrapurohit) [inline CRM/Price/BAO/PriceField.php]: I think we need to also add `description` class to make the span look similar to other help texts(custom fields, etc).

**Comment by jitendrapurohit** [comment]
Can you update this PR as it has gone stale due to generated data changes.

**Comment by JoeMurray** [comment]
Heh Seamus, as Release Manager this month, I'm trying to recruit people to help pare down the backlog of almost 100 PRs, some going back to last summer. I'm wondering if you would be able to help QA another PR if I got someone to QA this PR and your other ones like #7499 #8503 #8273 #8424 ?

File: CRM/Price/BAO/PriceField.php [hunk 2]
```diff
@@ -527,12 +530,16 @@ public static function addQuickFormElement(
           $max_value = CRM_Utils_Array::value('max_value', $opt, '');
 
           if ($field->is_display_amounts) {
+            $pre_help = isset($opt['help_pre']) ?
+              '<span class="crm-price-amount-help-pre">' . $opt['help_pre'] . '</span>: ' : '';
+            $post_help = isset($opt['help_post']) ?
+              ': <span class="crm-price-amount-help-post">' . $opt['help_post'] . '</span>' : '';
```
Comment (jitendrapurohit) [inline CRM/Price/BAO/PriceField.php]: @seamuslee001 shouldn't `description` class be added here too ?

**Comment by jitendrapurohit** [comment]
@seamuslee001 I get an  error on upgrade after the updated code.

>ALTER TABLE `civicrm_price_field_value_en_US` ADD COLUMN `help_pre_en_US` text COLLATE utf8_unicode_ci COMMENT "Price field option pre help text." [nativecode=1347 ** 'test_drupal_civi.civicrm_price_field_value_en_US' is not BASE TABLE]

Screenshot - http://goo.gl/Zds5N5

**Comment by JoeMurray** [comment]
@mlutfy could you review where the i18n / l10n is going wrong in patch?

**Comment by eileenmcnaughton** [comment]
I think you need to look at how you are calling executeQuery - here is the call sig

```
  public static function &executeQuery(
    $query,
    $params = array(),
    $abort = TRUE,
    $daoName = NULL,
    $freeDAO = FALSE,
    $i18nRewrite = TRUE,
    $trapException = FALSE
```

**Comment by seamuslee001** [comment]
thanks Eileen :) @eileenmcnaughton

**Comment by jitendrapurohit** [comment]
@seamuslee001 After multilingual upgrade to 4.7.9 with this patch included, fields added here(`pre help` and `post help`) do appear on the form, but they are not saved when the form is submitted.

Can you please check if this is replicated by you as well ?

Thanks!

**Comment by seamuslee001** [comment]
@jitendrapurohit I don't have access to a multilingual DB so bit difficult for me but I have just pushed something to try and make it like we're adding label. I don't know what is causing the problem

File: CRM/Price/Form/Option.php [hunk 3]
```diff
@@ -216,6 +216,8 @@ public function buildQuickForm() {
       $this->addRule('amount', ts('Please enter a monetary value for this field.'), 'money');
 
       $this->add('textarea', 'description', ts('Description'));
+      $this->add('textarea', 'help_pre', ts('Pre Option Help'), NULL, TRUE);
```
Comment (eileenmcnaughton) [inline CRM/Price/Form/Option.php]: I'm guessing the problem is in the postProcess on this file? You shouldn't need to do anything re multilingual - esp if you use the api

**Comment by jitendrapurohit** [comment]
Nope, it still occurs for me :-(. I see that altering `civicrm_price_field_value` in upgrade doesn't add these columns to `civicrm_price_field_value_{locale}` view (is these because of executeQuery() call?). 

The [PriceFieldValue takes the BAO object](https://github.com/civicrm/civicrm-core/blob/master/CRM/Price/BAO/PriceFieldValue.php#L54-L55) which doesn't have `pre_help` and `post_help` and copy the values from `$params` which fails to save these two fields.

**Comment by seamuslee001** [comment]
@jitendrapurohit Can you test it again please, I have tried again with the upgrade but not sure what else I can do

**Comment by jitendrapurohit** [comment]
Confirmed that this works fine now. I'll create a new PR for some minor fixes on display and optimization.

Thanks @seamuslee001 :-)

**Comment by jitendrapurohit** [comment]
Additional PR at https://github.com/civicrm/civicrm-core/pull/8642

**Pull request merged**